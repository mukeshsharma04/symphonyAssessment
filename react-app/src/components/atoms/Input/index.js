import React from "react";
import styled from "@emotion/styled";

// styles
import styles from "./input.styles";

const InputConatiner = styled("input")(styles.input);
const handleInputChange = (event, callback) => {
  event.preventDefault();
  callback(event.target.value);
};

const handleFileInput = (event, callback) => {
  event.preventDefault();
  callback(event.target.files);
};

const Input = ({ type, name, changeHandler }) => (
  <InputConatiner
    type={type}
    name={name}
    onChange={event =>
      type === "file"
        ? handleFileInput(event, changeHandler)
        : handleInputChange(event, changeHandler)
    }
  />
);

export default Input;
