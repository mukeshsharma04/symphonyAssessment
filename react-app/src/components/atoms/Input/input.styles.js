const styles = {
  input: {
    padding: "1px",
    margin: "10px"
  }
};

export default styles;
