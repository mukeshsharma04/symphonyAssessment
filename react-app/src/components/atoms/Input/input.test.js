import React from "react";
import renderer from "react-test-renderer";
import { cleanup, fireEvent, render } from "@testing-library/react";
import Input from "./";

// Note: running cleanup afterEach is done automatically for you in @testing-library/react@9.0.0 or higher
// unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

describe("Input", () => {
  const mockeInputHandler = jest.fn();
  const mockFileTypeHandler = jest.fn();

  test("render correctly", () => {
    const mockProps = {
      type: "text",
      name: "name",
      changeHandler: mockeInputHandler
    };
    let wrapper = renderer.create(<Input {...mockProps} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  test("should handle input type file", () => {
    const mockProps = {
      type: "file",
      name: "fileInput",
      changeHandler: mockFileTypeHandler
    };
    const mockFile = new File(["sample"], "sample.txt", {
      type: "text/plain"
    });
    const { container } = render(<Input {...mockProps} />);
    const input = container.firstChild;
    fireEvent.change(input, { target: { files: [mockFile] } });

    expect(mockFileTypeHandler).toHaveBeenCalledTimes(1);
  });

  test("should handle input type text", () => {
    const { container } = render(
      <Input type="text" changeHandler={mockeInputHandler} />
    );
    const input = container.firstChild;
    fireEvent.change(input, { target: { value: "value" } });
    expect(mockeInputHandler).toHaveBeenCalledTimes(1);
    expect(input.value).toEqual("value");
  });
});
