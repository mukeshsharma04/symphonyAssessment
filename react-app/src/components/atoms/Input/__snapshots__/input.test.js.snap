// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render correctly 1`] = `
<input
  className="css-1nvkxsy"
  name="name"
  onChange={[Function]}
  type="text"
/>
`;
